refactor: streamlined check_index_exists and removed redundancy

harikrishnatp opened the pull request:
<!---
Thank you for your pull request! 🚀
-->

### Contributor checklist

<!-- Please replace the empty checkboxes [] below with checked ones [x] accordingly. -->

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch
- [x] I have tested my code with the `pytest` command as directed in the [testing section of the contributing guide](https://github.com/scribe-org/Scribe-Data/blob/main/CONTRIBUTING.md#testing)

---

### Description

As duplicates of existing files check existed in different files, it was not efficient and caused redundancy. Applied new logic to check_index_exists to adapt to different cases and removed the redundant code. I have written tests for different scenarios and manually tested it as well.

### Related issue

#537

andrewtavis commented:
Thanks so much for the PR, @harikrishnatp! Really appreciate you keeping this in mind :) Would you be able to do an initial review here, @axif0? :)

axif0 commented:
I'm getting this error, when I try 
```
scribe-data g -a -dt verbs -od test
```

![image](https://github.com/user-attachments/assets/6de48eb2-4925-471f-9b2a-622c38ca8c34)

andrewtavis commented:
Maybe there's an issue with the way that the default query is being set? Generally it checks to see if there's a `.sparql` file with `*_1.sparql` and then it doesn't check for the one that doesn't have a number at the end 🤔

andrewtavis commented:
Quick check in here, @harikrishnatp :) Just want to make sure that you have the support you need for the further work here 😊 Please let us know if you have any questions or if we can help somehow!

harikrishnatp commented:
Thank you for checking in ! I've been occupied with a few things for some days. I had a hackathon to attend so for that, I had to make some preparations. Will get back to this as soon as possible!

andrewtavis commented:
Sounds great, @harikrishnatp! I remember the hackathon now :) Hope it went well! 😊

andrewtavis commented:
Nice! Quite the achievement all the same 😊

harikrishnatp commented:
> Maybe there's an issue with the way that the default query is being set? Generally it checks to see if there's a `.sparql` file with `*_1.sparql` and then it doesn't check for the one that doesn't have a number at the end 🤔

Yes, the problem was with this one. I modified the code in a way that it first checks if non numbered files exists and then for numbered ones without replacing. I don't know how it caused problem but now I've changed it

harikrishnatp commented:
```(venv) (base) hkx05@harikrishna:~/Scribe-Data$ scribe-data get -lang Malayalam -dt nouns -i
? What would you like to do? Configure get data request
Follow the prompts below. Press tab for completions and enter to select.
Select languages (comma-separated or 'All'): german
Select data types (comma-separated or 'All'): nouns
Select output type (json/csv/tsv): json
Enter output directory (default: scribe_data_json_export):
Overwrite existing files? (Y/n): y


  Scribe-Data Request Configuration Summary                                                                                                                            
┏━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━┓                                                                                                                         
┃ Setting          ┃ Value(s)                ┃                                                                                                                         
┡━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━┩                                                                                                                         
│ Languages        │ german                  │                                                                                                                         
│ Data Types       │ nouns                   │                                                                                                                         
│ Output Type      │ json                    │                                                                                                                         
│ Output Directory │ scribe_data_json_export │                                                                                                                         
│ Overwrite        │ Yes                     │                                                                                                                         
└──────────────────┴─────────────────────────┘                                                                                                                         


? What would you like to do? Request for get data
Exporting german nouns data:   0%|                                                                                                        | 0/1 [00:00<?, ?operation/s]Querying and formatting German nouns
Wrote file german/nouns.json with 164,912 nouns.
Successfully queried and formatted data for German nouns.
Updated data was saved in: /home/hkx05/Scribe-Data/scribe_data_json_export.
[18:32:41] INFO     ✘ Failed to export german nouns data.                                                                                            interactive.py:232
Exporting german nouns data: 100%|████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [01:30<00:00, 90.95s/operation]
Data request completed successfully!
Thank you for using Scribe-Data!
Existing file(s) found for Malayalam and Nouns in scribe_data_json_export.
1. nouns.json

Overwrite existing data? (o for overwrite, any other key to skip): 
Skipping update for Malayalam nouns.```

I suppose interactive is messed up ?

andrewtavis commented:
Looks like it 🤔 But some of this could be fixed on main. Let's figure out the merge conflicts and integrate the current main and then we can go from there 😊

axif0 commented:
> (venv) (base) hkx05@harikrishna:~/Scribe-Data$ scribe-data get -lang Malayalam -dt nouns -i

Shouldn't correct command be 
```
scribe-data get -i
```
By adding `-i`, Interactive mode we must need to give language, and data_types.
![image](https://github.com/user-attachments/assets/3ca9c529-3c8b-47d0-a109-636240465b24)

And yes, you should marge your branch with main.

harikrishnatp commented:
> > (venv) (base) hkx05@harikrishna:~/Scribe-Data$ scribe-data get -lang Malayalam -dt nouns -i
> 
> Shouldn't correct command be
> 
> ```
> scribe-data get -i
> ```
> 
> By adding `-i`, Interactive mode we must need to give language, and data_types. ![image](https://private-user-images.githubusercontent.com/66010854/419467514-3ca9c529-3c8b-47d0-a109-636240465b24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDExODg3OTUsIm5iZiI6MTc0MTE4ODQ5NSwicGF0aCI6Ii82NjAxMDg1NC80MTk0Njc1MTQtM2NhOWM1MjktM2M4Yi00N2QwLWExMDktNjM2MjQwNDY1YjI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA1VDE1MjgxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkwYWIzYTAzZWU2OTNmZmY0YzA5Yzg1YjlhYTUzODA4MjJjZGIyZDE0OWVlMDUxZDEwYjQzMmYwM2FmYWQ3NjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DnBxdZpUG3UTb1g0RJKabpj6jm2Dt6YcrlqNT0gZ9Rk)
> 
> And yes, you should marge your branch with main.
> > (venv) (base) hkx05@harikrishna:~/Scribe-Data$ scribe-data get -lang Malayalam -dt nouns -i
> 
> Shouldn't correct command be
> 
> ```
> scribe-data get -i
> ```
> 
> By adding `-i`, Interactive mode we must need to give language, and data_types. ![image](https://private-user-images.githubusercontent.com/66010854/419467514-3ca9c529-3c8b-47d0-a109-636240465b24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDExODg3OTUsIm5iZiI6MTc0MTE4ODQ5NSwicGF0aCI6Ii82NjAxMDg1NC80MTk0Njc1MTQtM2NhOWM1MjktM2M4Yi00N2QwLWExMDktNjM2MjQwNDY1YjI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA1VDE1MjgxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkwYWIzYTAzZWU2OTNmZmY0YzA5Yzg1YjlhYTUzODA4MjJjZGIyZDE0OWVlMDUxZDEwYjQzMmYwM2FmYWQ3NjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DnBxdZpUG3UTb1g0RJKabpj6jm2Dt6YcrlqNT0gZ9Rk)
> 
> And yes, you should marge your branch with main.

Yeah, I thought I was using -i wrong. I also merged with main—almost lost my code changes, but thank god I stashed them
Yeah, I had the thought that I was using -i wrong. I also merged with main. Almost lost my code changes but thank god I stashed them

andrewtavis commented:
Let's definitely push the merged branch and then we can go from there on finalizing this 😊

andrewtavis commented:
Just checking that the plan is to reopen here, right @harikrishnatp? :)

andrewtavis commented:
Nice @harikrishnatp! Let us know when it's ready to review :)

@axif0: Can you do a first review and I'll finalize it? 😊

harikrishnatp commented:
It's ready to be reviewed @andrewtavis @axif0

axif0 commented:
In interactive mood, the `check_index_exists` function get triggered, maybe we can pass a parameter as `interactive=True/False`, So that the message won't show. 
![image](https://github.com/user-attachments/assets/32005ab1-5945-4dd6-9423-3297c75b6e0f)


@harikrishnatp, Sorry for the late, Overall nice work.

andrewtavis commented:
Do you want to send along the change so we can merge this, @axif0? Might be easier and then @harikrishnatp can pick up another issue :)
